src/navigation/hierarchyMixin.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+const Poller = require('ft-poller');
+const ms = require('ms');
+
+function clone(obj){
+	return JSON.parse(JSON.stringify(obj));
+}
+
+function findItem(data, id, parent){
+	for(let item of data){
+		if(item.id === id){
+			return {
+				item: item,
+				parent: parent
+			}
+		}
+
+		if(item.children){
+			return findItem(item.children, id, item);
+		}
+	}
+
+	return {item:null, parent:null};
+}
+
+module.exports = class HierarchyMixin {
+
+	constructor(){
+		this.poller = new Poller({
+			url: 'http://next-navigation.ft.com/v1/taxonomy',
+			refreshInterval: ms('15m')
```

Comment:
I wouldn't use it client-side but on the server why now?  It brings in no other dependencies and makes the code a bit easier to understand